### ISSUE TITLE
Currency in bank ledger, general ledger is mixed when a foreign currency account is used

### ISSUE THREAD

**Issue by aurelienmi** [Dolibarr/dolibarr#17462]
# Bug
[If we create a bank account with a foreign currency (for example in Japanese Yen, or US dollars, the total view of all bank writings is mixing the different currency and the balance is then wrong because the amounts are not converted correctly]

## Environment
- **Version**: tested on Dolibarr v12, and same issue in Dolibarr v13
- **OS**: Windows
- **Web server**: Apache
- **PHP**: PHP 7.3, PHP 7.4
- **Database**: MySQL and PostgreSQL
- **URL(s)**: 
htdocs/compta/bank/bankentries_list.php?leftmenu= (global bank writings)
htdocs/accountancy/journal/bankjournal.php?id_journal=3 (bank ledger)

## Expected and actual behavior
Actual behavior : on those bank writings, we can see that all writings are not converted and so we don't have a coherent sum
Expected behavior : on the individual table writings for one bank account, foreign currencies can be used, but on those global tables every amount has to be converted correctly in EUR to have a correct sum, a correct bank ledger, and a correct general ledger

## Steps to reproduce the behavior
1°) create a bank account with a foreign currency
2°) for example create a supplier invoice in this foreign currency
3°) check that the global bank writings is wrong
4°) populate the bank ledger and / or the general ledger with the corresponding writings
5°) check that the bank ledger and general ledger are not converted in EUR to have coherent view and account ledger

## [Attached files](https://help.github.com/articles/issue-attachments) (Screenshots, screencasts, dolibarr.log, debugging informations…)
![image](https://user-images.githubusercontent.com/33426521/116814708-8fa08480-ab5a-11eb-8c08-924cd81260fe.png)

**Comment by soundheritage** [comment]
I have the same issue. I will be avalable to do more test on a dolibarr with postgreSQL database

**Comment by rhoizey** [comment]
Hi,

I face the same issue.
Th precise situation is the following : 
  - My main currency is XPF (I'm located in Tahiti and my business is mainly in XPF).
  - My bank account is in France, so it's in EUR.
  - I register a supplier order in XPF (supplier's currency is XPF).
  - I register a payment. The amount is in XPF.
  - On the bank account, I see the XPF amount debited from my EUR account without conversion.

The problem exists in v13, v14 and v15 alpha.

**Comment by soundheritage** [comment]
Hi, 
Can you give us a statut for this issue ?
Do you know when you can do a correction for this issue ?

I'm free to do test if you need, let me know

Thanks in advance

**Comment by aurelienmi** [comment]
Dear all,
maybe we can get a comment from the specialists of accounting with Dolibarr
For example it may be interesting to know if this issue may be fixed with the new Crowd Funding Campaign that has been made recently ?
Thanks a lot for your feedback

**Comment by soundheritage** [comment]
Dear All, 

Is there any correction regarding this issue ?
aurelienmi ask a good question, any feedback ?

Best regards

**Comment by soundheritage** [comment]
Did you have any feedback on this issue ?

**Comment by aurelienmi** [comment]
Dear all,

we are several with the same issue, we would like to have a feedback, if this issue is reproduced by other in the community ?
Thanks for your kind support on this specific issue of currency coherence.

**Comment by soundheritage** [comment]
I try to update to the last dolibarr 15 version, to check if the issue is solved with this update.
Unfortunately, this issue is still there. Can you check please, if there is a fast workaround ? 
because i need to extract my balance for my french administration, and all the figures are wrong...
Thanks for your feedback and your attention.

**Comment by artemChernitsov** [comment]
Hi all
I have the same issue
Debit transactions in multi-currencies bank accounts calculate in one currency but do not exchange rate.

**Comment by soundheritage** [comment]
Hello, 
Please, we are many to have this issue. Can you plan to do correction, or give us a workaround ?

**Comment by ileade** [comment]
Hi,

In version 17, the problem is still existing.
This is a huge problem as account follow-up is always false.
And for a multi-currency software such as dolibarr, it should be fixed quickly as it is not an "exotic" situation. 

Please, provide a fix. This is important.

**Comment by philchamp** [comment]
I have the same issue v16.0.3. I've been using dolibarr for 3 years and while doing my accounting for 2022 and 2023, I noticed this exact problem. It started in December 2022 shortly after I updated from 14.0.3 -> 16.0.3. In the 2 years prior to that, it was working well.

Edit: FYI, the account is Bank USD. My business is CAD. I have many CAD transactions that should be in USD in my USD Bank account. The suppliers are the same as they were 2 years ago with no changes. Payments and invoices are in USD.
